crates/orchestrator/src/tests/config.rs

```diff
@@ -64,116 +93,347 @@ impl TestConfigBuilder {
             queue: None,
             storage: None,
             alerts: None,
+
+            database_node: None,
+            data_storage_node: None,
+            data_storage_client: None,
+            queue_node: None,
+            queue_client: None,
+            alert_client: None,
+            alert_node: None,
         }
     }
 
-    pub fn mock_da_client(mut self, da_client: Box<dyn DaClient>) -> TestConfigBuilder {
+    pub async fn testcontainer_s3_data_storage(mut self) -> TestConfigBuilder {
+        let (node, storage_client, client) = s3_testcontainer_setup().await;
+        self.data_storage_node = Some(node);
+        self.data_storage_client = Some(client);
+        self.storage = Some(storage_client);
+        self
+    }
+
+    pub async fn testcontainer_sqs_queue(mut self) -> TestConfigBuilder {
+        let (node, queue_client, client) = sqs_testcontainer_setup().await;
+        self.queue = Some(queue_client);
+        self.queue_client = Some(client);
+        self.queue_node = Some(node);
+        self
+    }
+
+    // IMP! Don't use SQS ans SNS testcontainer setups together
+    pub async fn testcontainer_sns_sqs_alert(mut self) -> TestConfigBuilder {
+        let (node, sns_alert, sqs_queue, sns_client, sqs_client, _sqs_arn, _queue_host_url) =
+            sns_sqs_testcontainer_setup().await;
+        self.queue = Some(sqs_queue);
+        self.queue_client = Some(sqs_client);
+        self.alerts = Some(sns_alert);
+        self.alert_client = Some(sns_client);
+
+        self.alert_node = Some(node);
+        self
+    }
+
+    pub async fn testcontainer_mongo_database(mut self) -> TestConfigBuilder {
+        let (node, database) = mongodb_testcontainer_setup().await;
```

Comment:
i would call the `setup` functions during build and not during these intermediate builder functions

Reply:
Builders only ensures that if anything is not built then add a mock client to it, rest all the functions be it testcontainers or adding creates/takes services and stores in the builder, 
if we want to build them in `.build()` function then we'll have to maintain extra variables to manage state of what needs to be built and what needs to be mocked, I don't think it's worth the effort. 
WDYT ?

Reply:
discussed in person